opencensus/stats/aggregation_data.py

```diff
@@ -221,15 +243,62 @@ def __init__(self, value):
         super(LastValueAggregationData, self).__init__(value)
         self._value = value
 
-    def add_sample(self, value):
+    def add_sample(self, value, timestamp, attachments):
         """Adds a sample to the current
-           LastValue Aggregation Data and overwrite
-           the current recorded value
-        """
+        LastValue Aggregation Data and overwrite
+        the current recorded value"""
         self._value = value
 
     @property
     def value(self):
         """The current value recorded
         """
         return self._value
+
+
+class Exemplar(object):
+    """ Exemplar represents an example point that may be used to annotate
+        aggregated distribution values, associated with a histogram bucket.
+
+        :type value: double
+        :param value: value of the Exemplar point.
+
+        :type timestamp: time
+        :param timestamp: the time that this Exemplar's value was recorded.
+
+        :type attachments: dict
+        :param attachments: the contextual information about the example value.
+    """
+
+    def __init__(self,
+                 value,
+                 timestamp,
+                 attachments):
+        self._value = value
+
+        self._timestamp = timestamp
+
+        if attachments is None:
+            raise TypeError('attachments should not be empty')
+
+        for key, value in attachments.items():
+            if key is None:
```

Comment:
Currently we only use `string` as exemplar attachments. Here you can either check `key` and `value` are `string`, or do explicit `str` conversion on key and value.

Reply:
Done